doc/api/stream.md

```diff
@@ -1608,6 +1620,41 @@ async function run() {
 run().catch(console.error);
 ```
 
+```js
+const pipeline = util.promisify(stream.pipeline);
+const fs = require('fs').promises;
+
+async function run() {
+  await pipeline(
+    async function*() {
+      const fd = await fs.open('archive.tar');
+      try {
+        const chunk = new Buffer(1024);
+        const { bytesRead } = await fs.read(fd, chunk, 0, chunk.length, null);
+        if (bytesRead === 0) return;
+        yield chunk.slice(0, bytesRead);
+      } finally {
+        await fs.close(fd);
+      }
+    },
+    zlib.createGzip(),
+    async function(source) {
+      const fd = await fs.open('archive.tar', 'w');
+      try {
+        for await (const chunk of source) {
+          fs.write(fd, chunk);
+        }
+      } finally {
+        await fs.close(fd);
+      }
+    }
```

Comment:
Can we maybe note in the examples that the async function/async generator behave like fs streams, respectively? I am worried that this example might be encouraging manually writing fs streams this way.

Reply:
Hmm... in that case maybe we should think of a better example? Any ideas?